packages/web/next.config.js

```diff
@@ -27,6 +29,27 @@ const config = {
       commons: { chunks: "initial" },
     };
 
+    // Replace libsodium with a no-op API. It is only imported from within cosmJS to support
+    // argon2i and ed25519, both functionalities which in the context of Cosmos would only get used within
+    // an extension wallet. Libsodium is ~190kb gzipped, 500kb parsed, so this meaningfully reduces client load.
+    // (And it gets bundled twice)
+    //
+    // It should never be getting used. This is copied from what Keplr does:
+    // https://github.com/chainapsis/keplr-wallet/blob/master/package.json#L103-L104
+    config.resolve = {
+      ...config.resolve, // This spreads existing resolve configuration (if any)
+      alias: {
+        ...config.resolve.alias, // This spreads any existing alias configurations
+        libsodium: path.resolve(__dirname, "etc", "noop", "index.js"),
+        "libsodium-wrappers": path.resolve(
+          __dirname,
+          "etc",
+          "noop",
+          "index.js"
+        ),
```

Comment:
For more recent versions of CosmJS you need to patch out `libsodium-wrappers-sumo`. So ideally you just do the same thing for all four:

```
libsodium
libsodium-wrappers
libsodium-sumo
libsodium-wrappers-sumo
```

Reply:
Oh awesome, thank you!